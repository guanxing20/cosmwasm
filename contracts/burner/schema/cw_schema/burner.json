{
  "contract_name": "burner",
  "contract_version": "0.0.0",
  "idl_version": "1.0.0",
  "instantiate": {
    "type": "v1",
    "root": 0,
    "definitions": [
      {
        "name": "burner_msg_InstantiateMsg",
        "description": "A placeholder where we don't take any input",
        "type": "struct",
        "properties": {}
      }
    ]
  },
  "execute": null,
  "query": null,
  "migrate": {
    "type": "v1",
    "root": 0,
    "definitions": [
      {
        "name": "burner_msg_MigrateMsg",
        "type": "struct",
        "properties": {
          "delete": {
            "defaulting": true,
            "description": "Optional amount of items to delete in this call.\nIf it is not provided, nothing will be deleted.\nYou can delete further items in a subsequent execute call.",
            "value": 3
          },
          "denoms": {
            "description": "The denoms of the final payout. Balances of tokens not listed here\nwill remain in the account untouched.",
            "value": 2
          },
          "payout": {
            "description": "The address we send all remaining balance to. See denoms\nbelow for the denoms to consider.",
            "value": 1
          }
        }
      },
      {
        "name": "String",
        "type": "string"
      },
      {
        "name": "alloc::vec::Vec<alloc::string::String>",
        "type": "array",
        "items": 1
      },
      {
        "name": "u32",
        "type": "integer",
        "precision": 32,
        "signed": false
      }
    ]
  },
  "sudo": null,
  "responses": null
}
