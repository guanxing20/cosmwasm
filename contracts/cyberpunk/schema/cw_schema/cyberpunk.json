{
  "contract_name": "cyberpunk",
  "contract_version": "0.0.0",
  "idl_version": "1.0.0",
  "instantiate": {
    "type": "v1",
    "root": 0,
    "definitions": [
      {
        "name": "cosmwasm_std_results_empty_Empty",
        "description": "An empty struct that serves as a placeholder in different places,\nsuch as contracts that don't set a custom message.\n\nIt is designed to be expressible in correct JSON and JSON Schema but\ncontains no meaningful data. Previously we used enums without cases,\nbut those cannot represented as valid JSON Schema (https://github.com/CosmWasm/cosmwasm/issues/451)",
        "type": "struct",
        "properties": {}
      }
    ]
  },
  "execute": {
    "type": "v1",
    "root": 0,
    "definitions": [
      {
        "name": "cyberpunk_msg_ExecuteMsg",
        "type": "enum",
        "cases": {
          "allocate_large_memory": {
            "description": "Allocate large amounts of memory without consuming much gas",
            "type": "named",
            "properties": {
              "pages": {
                "value": 1
              }
            }
          },
          "argon2": {
            "description": "Hashes some data. Uses CPU and memory, but no external calls.",
            "type": "named",
            "properties": {
              "mem_cost": {
                "description": "The amount of memory requested (KB).",
                "value": 1
              },
              "time_cost": {
                "description": "The number of passes.",
                "value": 1
              }
            }
          },
          "cpu_loop": {
            "description": "Infinite loop to burn cpu cycles (only run when metering is enabled)",
            "type": "named",
            "properties": {}
          },
          "debug": {
            "description": "Does a bit of work and calls debug",
            "type": "named",
            "properties": {}
          },
          "memory_loop": {
            "description": "Infinite loop reading and writing memory",
            "type": "named",
            "properties": {}
          },
          "message_loop": {
            "description": "Infinite loop sending message to itself",
            "type": "named",
            "properties": {}
          },
          "mirror_env": {
            "description": "Returns the env for testing",
            "type": "named",
            "properties": {}
          },
          "noop": {
            "description": "Does nothing. This can be used for baseline contract execution performance measurements.",
            "type": "named",
            "properties": {}
          },
          "panic": {
            "description": "Trigger a panic to ensure framework handles gracefully",
            "type": "named",
            "properties": {}
          },
          "storage_loop": {
            "description": "Infinite loop making storage calls (to test when their limit hits)",
            "type": "named",
            "properties": {}
          },
          "unreachable": {
            "description": "In contrast to Panic, this does not use the panic handler.\n\nFrom <https://doc.rust-lang.org/beta/core/arch/wasm32/fn.unreachable.html>:\n\"Generates the unreachable instruction, which causes an unconditional trap.\"",
            "type": "named",
            "properties": {}
          }
        }
      },
      {
        "name": "u32",
        "type": "integer",
        "precision": 32,
        "signed": false
      }
    ]
  },
  "query": {
    "type": "v1",
    "root": 0,
    "definitions": [
      {
        "name": "cyberpunk_msg_QueryMsg",
        "type": "enum",
        "cases": {
          "denom": {
            "description": "Queries `DenomMetadata` from the bank module and returns the result",
            "type": "named",
            "properties": {
              "denom": {
                "value": 1
              }
            }
          },
          "denoms": {
            "description": "Queries `AllDenomMetadata` from the bank module repeatedly and returns all entries",
            "type": "named",
            "properties": {}
          },
          "mirror_env": {
            "description": "Returns the env for testing",
            "type": "named",
            "properties": {}
          }
        }
      },
      {
        "name": "String",
        "type": "string"
      }
    ]
  },
  "migrate": null,
  "sudo": null,
  "responses": {
    "denom": {
      "type": "v1",
      "root": 0,
      "definitions": [
        {
          "name": "cosmwasm_std_metadata_DenomMetadata",
          "description": "Replicates the cosmos-sdk bank module Metadata type",
          "type": "struct",
          "properties": {
            "base": {
              "value": 1
            },
            "denom_units": {
              "value": 5
            },
            "description": {
              "value": 1
            },
            "display": {
              "value": 1
            },
            "name": {
              "value": 1
            },
            "symbol": {
              "value": 1
            },
            "uri": {
              "value": 1
            },
            "uri_hash": {
              "value": 1
            }
          }
        },
        {
          "name": "String",
          "type": "string"
        },
        {
          "name": "cosmwasm_std_metadata_DenomUnit",
          "description": "Replicates the cosmos-sdk bank module DenomUnit type",
          "type": "struct",
          "properties": {
            "aliases": {
              "value": 4
            },
            "denom": {
              "value": 1
            },
            "exponent": {
              "value": 3
            }
          }
        },
        {
          "name": "u32",
          "type": "integer",
          "precision": 32,
          "signed": false
        },
        {
          "name": "alloc::vec::Vec<alloc::string::String>",
          "type": "array",
          "items": 1
        },
        {
          "name": "alloc::vec::Vec<cosmwasm_std::metadata::DenomUnit>",
          "type": "array",
          "items": 2
        }
      ]
    },
    "denoms": {
      "type": "v1",
      "root": 6,
      "definitions": [
        {
          "name": "cosmwasm_std_metadata_DenomMetadata",
          "description": "Replicates the cosmos-sdk bank module Metadata type",
          "type": "struct",
          "properties": {
            "base": {
              "value": 1
            },
            "denom_units": {
              "value": 5
            },
            "description": {
              "value": 1
            },
            "display": {
              "value": 1
            },
            "name": {
              "value": 1
            },
            "symbol": {
              "value": 1
            },
            "uri": {
              "value": 1
            },
            "uri_hash": {
              "value": 1
            }
          }
        },
        {
          "name": "String",
          "type": "string"
        },
        {
          "name": "cosmwasm_std_metadata_DenomUnit",
          "description": "Replicates the cosmos-sdk bank module DenomUnit type",
          "type": "struct",
          "properties": {
            "aliases": {
              "value": 4
            },
            "denom": {
              "value": 1
            },
            "exponent": {
              "value": 3
            }
          }
        },
        {
          "name": "u32",
          "type": "integer",
          "precision": 32,
          "signed": false
        },
        {
          "name": "alloc::vec::Vec<alloc::string::String>",
          "type": "array",
          "items": 1
        },
        {
          "name": "alloc::vec::Vec<cosmwasm_std::metadata::DenomUnit>",
          "type": "array",
          "items": 2
        },
        {
          "name": "alloc::vec::Vec<cosmwasm_std::metadata::DenomMetadata>",
          "type": "array",
          "items": 0
        }
      ]
    },
    "mirror_env": {
      "type": "v1",
      "root": 0,
      "definitions": [
        {
          "name": "cosmwasm_std_types_Env",
          "type": "struct",
          "properties": {
            "block": {
              "value": 1
            },
            "contract": {
              "value": 9
            },
            "transaction": {
              "description": "Information on the transaction this message was executed in.\nThe field is unset when the `MsgExecuteContract`/`MsgInstantiateContract`/`MsgMigrateContract`\nis not executed as part of a transaction.",
              "value": 8
            }
          }
        },
        {
          "name": "cosmwasm_std_types_BlockInfo",
          "type": "struct",
          "properties": {
            "chain_id": {
              "value": 4
            },
            "height": {
              "description": "The height of a block is the number of blocks preceding it in the blockchain.",
              "value": 2
            },
            "time": {
              "description": "Absolute time of the block creation in seconds since the UNIX epoch (00:00:00 on 1970-01-01 UTC).\n\nThe source of this is the [BFT Time in Tendermint](https://github.com/tendermint/tendermint/blob/58dc1726/spec/consensus/bft-time.md),\nwhich has the same nanosecond precision as the `Timestamp` type.\n\n# Examples\n\nUsing chrono:\n\n```\n# use cosmwasm_std::{Addr, Binary, BlockInfo, ContractInfo, Env, MessageInfo, Timestamp, TransactionInfo};\n# let env = Env {\n#     block: BlockInfo {\n#         height: 12_345,\n#         time: Timestamp::from_nanos(1_571_797_419_879_305_533),\n#         chain_id: \"cosmos-testnet-14002\".to_string(),\n#     },\n#     transaction: Some(TransactionInfo::new(3, Binary::from_hex(\"E5469DACEC17CEF8A260FD37675ED87E7FB6A2B5AD95193C51308006C7E494B3\").unwrap())),\n#     contract: ContractInfo {\n#         address: Addr::unchecked(\"contract\"),\n#     },\n# };\n# extern crate chrono;\nuse chrono::NaiveDateTime;\nlet seconds = env.block.time.seconds();\nlet nsecs = env.block.time.subsec_nanos();\nlet dt = NaiveDateTime::from_timestamp(seconds as i64, nsecs as u32);\n```\n\nCreating a simple millisecond-precision timestamp (as used in JavaScript):\n\n```\n# use cosmwasm_std::{Addr, Binary, BlockInfo, ContractInfo, Env, MessageInfo, Timestamp, TransactionInfo};\n# let env = Env {\n#     block: BlockInfo {\n#         height: 12_345,\n#         time: Timestamp::from_nanos(1_571_797_419_879_305_533),\n#         chain_id: \"cosmos-testnet-14002\".to_string(),\n#     },\n#     transaction: Some(TransactionInfo::new(3, Binary::from_hex(\"E5469DACEC17CEF8A260FD37675ED87E7FB6A2B5AD95193C51308006C7E494B3\").unwrap())),\n#     contract: ContractInfo {\n#         address: Addr::unchecked(\"contract\"),\n#     },\n# };\nlet millis = env.block.time.nanos() / 1_000_000;\n```",
              "value": 3
            }
          }
        },
        {
          "name": "u64",
          "type": "integer",
          "precision": 64,
          "signed": false
        },
        {
          "name": "cosmwasm_std_timestamp_Timestamp",
          "description": "A point in time in nanosecond precision.\n\nThis type can represent times from 1970-01-01T00:00:00Z to 2554-07-21T23:34:33Z.\n\n## Examples\n\n```\n# use cosmwasm_std::Timestamp;\nlet ts = Timestamp::from_nanos(1_000_000_202);\nassert_eq!(ts.nanos(), 1_000_000_202);\nassert_eq!(ts.seconds(), 1);\nassert_eq!(ts.subsec_nanos(), 202);\n\nlet ts = ts.plus_seconds(2);\nassert_eq!(ts.nanos(), 3_000_000_202);\nassert_eq!(ts.seconds(), 3);\nassert_eq!(ts.subsec_nanos(), 202);\n```",
          "type": "timestamp"
        },
        {
          "name": "String",
          "type": "string"
        },
        {
          "name": "cosmwasm_std_types_TransactionInfo",
          "type": "struct",
          "properties": {
            "hash": {
              "defaulting": true,
              "description": "Hash of the transaction.\n\nIf the blockchain's CosmWasm version is below 3.0, this field\nwill default to being empty.",
              "value": 7
            },
            "index": {
              "description": "The position of this transaction in the block. The first\ntransaction has index 0.\n\nThis allows you to get a unique transaction identifier in this chain\nusing the pair (`env.block.height`, `env.transaction.index`).\n",
              "value": 6
            }
          }
        },
        {
          "name": "u32",
          "type": "integer",
          "precision": 32,
          "signed": false
        },
        {
          "name": "cosmwasm_std_binary_Binary",
          "description": "Binary is a wrapper around Vec<u8> to add base64 de/serialization\nwith serde. It also adds some helper methods to help encode inline.\n\nThis is only needed as serde-json-{core,wasm} has a horrible encoding for Vec<u8>.\nSee also <https://github.com/CosmWasm/cosmwasm/blob/main/docs/MESSAGE_TYPES.md>.",
          "type": "binary"
        },
        {
          "name": "core::option::Option<cosmwasm_std::types::TransactionInfo>",
          "type": "optional",
          "inner": 5
        },
        {
          "name": "cosmwasm_std_types_ContractInfo",
          "type": "struct",
          "properties": {
            "address": {
              "value": 10
            }
          }
        },
        {
          "name": "cosmwasm_std_addresses_Addr",
          "description": "A human readable address.\n\nIn Cosmos, this is typically bech32 encoded. But for multi-chain smart contracts no\nassumptions should be made other than being UTF-8 encoded and of reasonable length.\n\nThis type represents a validated address. It can be created in the following ways\n1. Use `Addr::unchecked(input)`\n2. Use `let checked: Addr = deps.api.addr_validate(input)?`\n3. Use `let checked: Addr = deps.api.addr_humanize(canonical_addr)?`\n4. Deserialize from JSON. This must only be done from JSON that was validated before\nsuch as a contract's state. `Addr` must not be used in messages sent by the user\nbecause this would result in unvalidated instances.\n\nThis type is immutable. If you really need to mutate it (Really? Are you sure?), create\na mutable copy using `let mut mutable = Addr::to_string()` and operate on that `String`\ninstance.",
          "type": "address"
        }
      ]
    }
  }
}
