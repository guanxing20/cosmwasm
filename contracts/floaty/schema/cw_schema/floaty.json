{
  "contract_name": "floaty",
  "contract_version": "0.0.0",
  "idl_version": "1.0.0",
  "instantiate": {
    "type": "v1",
    "root": 0,
    "definitions": [
      {
        "name": "cosmwasm_std_results_empty_Empty",
        "description": "An empty struct that serves as a placeholder in different places,\nsuch as contracts that don't set a custom message.\n\nIt is designed to be expressible in correct JSON and JSON Schema but\ncontains no meaningful data. Previously we used enums without cases,\nbut those cannot represented as valid JSON Schema (https://github.com/CosmWasm/cosmwasm/issues/451)",
        "type": "struct",
        "properties": {}
      }
    ]
  },
  "execute": null,
  "query": {
    "type": "v1",
    "root": 0,
    "definitions": [
      {
        "name": "floaty_msg_QueryMsg",
        "type": "enum",
        "cases": {
          "instructions": {
            "description": "Returns a list of all instructions",
            "type": "named",
            "properties": {}
          },
          "random_args_for": {
            "description": "Returns valid random arguments for the given instruction",
            "type": "named",
            "properties": {
              "instruction": {
                "value": 1
              },
              "seed": {
                "value": 2
              }
            }
          },
          "run": {
            "description": "Runs the given instruction with the given arguments and returns the result",
            "type": "named",
            "properties": {
              "args": {
                "value": 5
              },
              "instruction": {
                "value": 1
              }
            }
          }
        }
      },
      {
        "name": "String",
        "type": "string"
      },
      {
        "name": "u64",
        "type": "integer",
        "precision": 64,
        "signed": false
      },
      {
        "name": "floaty_instructions_Value",
        "type": "enum",
        "cases": {
          "f32": {
            "type": "tuple",
            "items": [
              4
            ]
          },
          "f64": {
            "type": "tuple",
            "items": [
              2
            ]
          },
          "u32": {
            "type": "tuple",
            "items": [
              4
            ]
          },
          "u64": {
            "type": "tuple",
            "items": [
              2
            ]
          }
        }
      },
      {
        "name": "u32",
        "type": "integer",
        "precision": 32,
        "signed": false
      },
      {
        "name": "alloc::vec::Vec<floaty::instructions::Value>",
        "type": "array",
        "items": 3
      }
    ]
  },
  "migrate": null,
  "sudo": null,
  "responses": {
    "instructions": {
      "type": "v1",
      "root": 1,
      "definitions": [
        {
          "name": "String",
          "type": "string"
        },
        {
          "name": "alloc::vec::Vec<alloc::string::String>",
          "type": "array",
          "items": 0
        }
      ]
    },
    "random_args_for": {
      "type": "v1",
      "root": 3,
      "definitions": [
        {
          "name": "floaty_instructions_Value",
          "type": "enum",
          "cases": {
            "f32": {
              "type": "tuple",
              "items": [
                1
              ]
            },
            "f64": {
              "type": "tuple",
              "items": [
                2
              ]
            },
            "u32": {
              "type": "tuple",
              "items": [
                1
              ]
            },
            "u64": {
              "type": "tuple",
              "items": [
                2
              ]
            }
          }
        },
        {
          "name": "u32",
          "type": "integer",
          "precision": 32,
          "signed": false
        },
        {
          "name": "u64",
          "type": "integer",
          "precision": 64,
          "signed": false
        },
        {
          "name": "alloc::vec::Vec<floaty::instructions::Value>",
          "type": "array",
          "items": 0
        }
      ]
    },
    "run": {
      "type": "v1",
      "root": 0,
      "definitions": [
        {
          "name": "floaty_instructions_Value",
          "type": "enum",
          "cases": {
            "f32": {
              "type": "tuple",
              "items": [
                1
              ]
            },
            "f64": {
              "type": "tuple",
              "items": [
                2
              ]
            },
            "u32": {
              "type": "tuple",
              "items": [
                1
              ]
            },
            "u64": {
              "type": "tuple",
              "items": [
                2
              ]
            }
          }
        },
        {
          "name": "u32",
          "type": "integer",
          "precision": 32,
          "signed": false
        },
        {
          "name": "u64",
          "type": "integer",
          "precision": 64,
          "signed": false
        }
      ]
    }
  }
}
