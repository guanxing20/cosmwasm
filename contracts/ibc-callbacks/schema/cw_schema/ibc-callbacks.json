{
  "contract_name": "ibc-callbacks",
  "contract_version": "0.0.0",
  "idl_version": "1.0.0",
  "instantiate": {
    "type": "v1",
    "root": 0,
    "definitions": [
      {
        "name": "cosmwasm_std_results_empty_Empty",
        "description": "An empty struct that serves as a placeholder in different places,\nsuch as contracts that don't set a custom message.\n\nIt is designed to be expressible in correct JSON and JSON Schema but\ncontains no meaningful data. Previously we used enums without cases,\nbut those cannot represented as valid JSON Schema (https://github.com/CosmWasm/cosmwasm/issues/451)",
        "type": "struct",
        "properties": {}
      }
    ]
  },
  "execute": {
    "type": "v1",
    "root": 0,
    "definitions": [
      {
        "name": "ibc_callbacks_msg_ExecuteMsg",
        "type": "enum",
        "cases": {
          "transfer": {
            "type": "named",
            "properties": {
              "callback_type": {
                "defaulting": true,
                "description": "Who should receive callbacks for the message",
                "value": 3
              },
              "channel_id": {
                "description": "The channel to send the packet through",
                "value": 1
              },
              "timeout_seconds": {
                "description": "The amount of seconds from now the transfer should timeout at",
                "value": 2
              },
              "to_address": {
                "description": "Address on the destination chain",
                "value": 1
              }
            }
          }
        }
      },
      {
        "name": "String",
        "type": "string"
      },
      {
        "name": "u32",
        "type": "integer",
        "precision": 32,
        "signed": false
      },
      {
        "name": "ibc_callbacks_msg_CallbackType",
        "type": "enum",
        "cases": {
          "both": {
            "description": "Both the source contract and the destination address should receive callbacks",
            "type": "unit"
          },
          "dst": {
            "description": "Only the destination address should receive callbacks",
            "type": "unit"
          },
          "src": {
            "description": "Only this contract on the source chain should receive callbacks",
            "type": "unit"
          }
        }
      }
    ]
  },
  "query": {
    "type": "v1",
    "root": 0,
    "definitions": [
      {
        "name": "ibc_callbacks_msg_QueryMsg",
        "type": "enum",
        "cases": {
          "callback_stats": {
            "description": "Returns stats about what callbacks have been received",
            "type": "named",
            "properties": {}
          }
        }
      }
    ]
  },
  "migrate": null,
  "sudo": null,
  "responses": {
    "callback_stats": {
      "type": "v1",
      "root": 0,
      "definitions": [
        {
          "name": "ibc_callbacks_state_CallbackStats",
          "description": "A counter for the number of times the respective callback has been called",
          "type": "struct",
          "properties": {
            "ibc_ack_callbacks": {
              "value": 14
            },
            "ibc_destination_callbacks": {
              "value": 23
            },
            "ibc_timeout_callbacks": {
              "value": 16
            }
          }
        },
        {
          "name": "cosmwasm_std_ibc_callbacks_IbcAckCallbackMsg",
          "type": "struct",
          "properties": {
            "acknowledgement": {
              "value": 2
            },
            "original_packet": {
              "value": 4
            },
            "relayer": {
              "value": 13
            }
          }
        },
        {
          "name": "cosmwasm_std_ibc_IbcAcknowledgement",
          "type": "struct",
          "properties": {
            "data": {
              "value": 3
            }
          }
        },
        {
          "name": "cosmwasm_std_binary_Binary",
          "description": "Binary is a wrapper around Vec<u8> to add base64 de/serialization\nwith serde. It also adds some helper methods to help encode inline.\n\nThis is only needed as serde-json-{core,wasm} has a horrible encoding for Vec<u8>.\nSee also <https://github.com/CosmWasm/cosmwasm/blob/main/docs/MESSAGE_TYPES.md>.",
          "type": "binary"
        },
        {
          "name": "cosmwasm_std_ibc_IbcPacket",
          "type": "struct",
          "properties": {
            "data": {
              "description": "The raw data sent from the other side in the packet",
              "value": 3
            },
            "dest": {
              "description": "identifies the channel and port on the receiving chain.",
              "value": 5
            },
            "sequence": {
              "description": "The sequence number of the packet on the given channel",
              "value": 7
            },
            "src": {
              "description": "identifies the channel and port on the sending chain.",
              "value": 5
            },
            "timeout": {
              "value": 8
            }
          }
        },
        {
          "name": "cosmwasm_std_ibc_IbcEndpoint",
          "type": "struct",
          "properties": {
            "channel_id": {
              "value": 6
            },
            "port_id": {
              "value": 6
            }
          }
        },
        {
          "name": "String",
          "type": "string"
        },
        {
          "name": "u64",
          "type": "integer",
          "precision": 64,
          "signed": false
        },
        {
          "name": "cosmwasm_std_ibc_IbcTimeout",
          "description": "In IBC each package must set at least one type of timeout:\nthe timestamp or the block height. Using this rather complex enum instead of\ntwo timeout fields we ensure that at least one timeout is set.",
          "type": "struct",
          "properties": {
            "block": {
              "value": 10
            },
            "timestamp": {
              "value": 12
            }
          }
        },
        {
          "name": "cosmwasm_std_ibc_IbcTimeoutBlock",
          "description": "IBCTimeoutHeight Height is a monotonically increasing data type\nthat can be compared against another Height for the purposes of updating and\nfreezing clients.\nOrdering is (revision_number, timeout_height)",
          "type": "struct",
          "properties": {
            "height": {
              "description": "block height after which the packet times out.\nthe height within the given revision",
              "value": 7
            },
            "revision": {
              "description": "the version that the client is currently on\n(e.g. after resetting the chain this could increment 1 as height drops to 0)",
              "value": 7
            }
          }
        },
        {
          "name": "core::option::Option<cosmwasm_std::ibc::IbcTimeoutBlock>",
          "type": "optional",
          "inner": 9
        },
        {
          "name": "cosmwasm_std_timestamp_Timestamp",
          "description": "A point in time in nanosecond precision.\n\nThis type can represent times from 1970-01-01T00:00:00Z to 2554-07-21T23:34:33Z.\n\n## Examples\n\n```\n# use cosmwasm_std::Timestamp;\nlet ts = Timestamp::from_nanos(1_000_000_202);\nassert_eq!(ts.nanos(), 1_000_000_202);\nassert_eq!(ts.seconds(), 1);\nassert_eq!(ts.subsec_nanos(), 202);\n\nlet ts = ts.plus_seconds(2);\nassert_eq!(ts.nanos(), 3_000_000_202);\nassert_eq!(ts.seconds(), 3);\nassert_eq!(ts.subsec_nanos(), 202);\n```",
          "type": "timestamp"
        },
        {
          "name": "core::option::Option<cosmwasm_std::timestamp::Timestamp>",
          "type": "optional",
          "inner": 11
        },
        {
          "name": "cosmwasm_std_addresses_Addr",
          "description": "A human readable address.\n\nIn Cosmos, this is typically bech32 encoded. But for multi-chain smart contracts no\nassumptions should be made other than being UTF-8 encoded and of reasonable length.\n\nThis type represents a validated address. It can be created in the following ways\n1. Use `Addr::unchecked(input)`\n2. Use `let checked: Addr = deps.api.addr_validate(input)?`\n3. Use `let checked: Addr = deps.api.addr_humanize(canonical_addr)?`\n4. Deserialize from JSON. This must only be done from JSON that was validated before\nsuch as a contract's state. `Addr` must not be used in messages sent by the user\nbecause this would result in unvalidated instances.\n\nThis type is immutable. If you really need to mutate it (Really? Are you sure?), create\na mutable copy using `let mut mutable = Addr::to_string()` and operate on that `String`\ninstance.",
          "type": "address"
        },
        {
          "name": "alloc::vec::Vec<cosmwasm_std::ibc::callbacks::IbcAckCallbackMsg>",
          "type": "array",
          "items": 1
        },
        {
          "name": "cosmwasm_std_ibc_callbacks_IbcTimeoutCallbackMsg",
          "type": "struct",
          "properties": {
            "packet": {
              "value": 4
            },
            "relayer": {
              "value": 13
            }
          }
        },
        {
          "name": "alloc::vec::Vec<cosmwasm_std::ibc::callbacks::IbcTimeoutCallbackMsg>",
          "type": "array",
          "items": 15
        },
        {
          "name": "cosmwasm_std_ibc_callbacks_IbcDestinationCallbackMsg",
          "description": "The message type of the IBC destination callback.\n\nThe IBC destination callback is needed for cases where someone triggers the sending of an\nIBC packet through some other message (i.e. not through [`crate::IbcMsg::SendPacket`]) and\nyour contract needs to know that it received this.\nA prominent example is the [`crate::IbcMsg::Transfer`] message. Without callbacks, you cannot know\nthat someone sent you IBC coins.\n\nIt is important to validate that the packet and acknowledgement are what you expect them to be.\nFor example for a transfer message, the receiver is not necessarily the contract itself.\n\nThe callback is called when the packet is being acknowledged on the destination chain.\nThis happens for both synchronous and asynchronous acknowledgements.\n\nNote that there are some prerequisites that need to be fulfilled to receive destination callbacks:\n- The contract must implement the `ibc_destination_callback` entrypoint.\n- The IBC application in the destination chain must have support for the callbacks middleware.\n- You have to add serialized [`IbcCallbackRequest`] to a specific field of the message.\nFor `IbcMsg::Transfer`, this is the `memo` field and it needs to be json-encoded.",
          "type": "struct",
          "properties": {
            "ack": {
              "value": 2
            },
            "packet": {
              "value": 4
            },
            "transfer": {
              "defaulting": true,
              "description": "When the underlying packet is a successful transfer message,\nthis field contains information about the transfer. Otherwise it is empty.\n\nThis is always empty on chains using CosmWasm < 3.0",
              "value": 22
            }
          }
        },
        {
          "name": "cosmwasm_std_ibc_callbacks_IbcTransferCallback",
          "type": "struct",
          "properties": {
            "funds": {
              "description": "The funds that were transferred.\n\nWhen the callback is executed, the transfer is completed already and the coins are now owned\nby the receiver.",
              "value": 21
            },
            "receiver": {
              "description": "Address of the receiver of the transfer.\nSince this is on the destination chain, this is a valid address.",
              "value": 13
            },
            "sender": {
              "description": "Address of the sender of the transfer.\nNote that this is *not* a valid address on the destination chain.",
              "value": 6
            }
          }
        },
        {
          "name": "cosmwasm_std_coin_Coin",
          "type": "struct",
          "properties": {
            "amount": {
              "value": 20
            },
            "denom": {
              "value": 6
            }
          }
        },
        {
          "name": "cosmwasm_std_math_uint256_Uint256",
          "description": "An implementation of u256 that is using strings for JSON encoding/decoding,\nsuch that the full u256 range can be used for clients that convert JSON numbers to floats,\nlike JavaScript and jq.\n\n# Examples\n\nUse `new` to create instances out of u128, `from` for other primitive uint types\nor `from_be_bytes` to provide big endian bytes:\n\n```\n# use cosmwasm_std::Uint256;\nlet a = Uint256::new(258u128);\nlet b = Uint256::from(258u16);\nlet c = Uint256::from_be_bytes([\n0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8,\n0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8,\n0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8,\n0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 1u8, 2u8,\n]);\nassert_eq!(a, b);\nassert_eq!(a, c);\n```",
          "type": "integer",
          "precision": 256,
          "signed": false
        },
        {
          "name": "alloc::vec::Vec<cosmwasm_std::coin::Coin>",
          "type": "array",
          "items": 19
        },
        {
          "name": "core::option::Option<cosmwasm_std::ibc::callbacks::IbcTransferCallback>",
          "type": "optional",
          "inner": 18
        },
        {
          "name": "alloc::vec::Vec<cosmwasm_std::ibc::callbacks::IbcDestinationCallbackMsg>",
          "type": "array",
          "items": 17
        }
      ]
    }
  }
}
