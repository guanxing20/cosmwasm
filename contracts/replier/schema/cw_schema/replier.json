{
  "contract_name": "replier",
  "contract_version": "0.0.0",
  "idl_version": "1.0.0",
  "instantiate": {
    "type": "v1",
    "root": 0,
    "definitions": [
      {
        "name": "replier_InstantiateMsg",
        "type": "struct",
        "properties": {}
      }
    ]
  },
  "execute": {
    "type": "v1",
    "root": 0,
    "definitions": [
      {
        "name": "replier_ExecuteMsg",
        "type": "struct",
        "properties": {
          "exec_error": {
            "value": 2
          },
          "messages": {
            "value": 3
          },
          "msg_id": {
            "value": 1
          },
          "reply_error": {
            "value": 2
          },
          "reply_on_never": {
            "value": 2
          },
          "return_order_in_reply": {
            "value": 2
          },
          "set_data_in_exec_and_reply": {
            "value": 2
          }
        }
      },
      {
        "name": "u8",
        "type": "integer",
        "precision": 8,
        "signed": false
      },
      {
        "name": "bool",
        "type": "boolean"
      },
      {
        "name": "alloc::vec::Vec<replier::ExecuteMsg>",
        "type": "array",
        "items": 0
      }
    ]
  },
  "query": {
    "type": "v1",
    "root": 0,
    "definitions": [
      {
        "name": "replier_QueryMsg",
        "type": "enum",
        "cases": {}
      }
    ]
  },
  "migrate": null,
  "sudo": null,
  "responses": {}
}
