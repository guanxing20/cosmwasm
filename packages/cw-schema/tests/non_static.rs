#![allow(dead_code)]

use cw_schema::Schemaifier;
use std::borrow::Cow;

#[derive(Schemaifier)]
struct NonStatic<'a> {
    test1: &'a str,
    test2: Cow<'a, str>,
    test3: Cow<'static, str>,
    test4: &'static str,
}

#[test]
fn non_static_schema() {
    let schema = cw_schema::schema_of::<NonStatic<'_>>();
    insta::assert_json_snapshot!(schema);
}
