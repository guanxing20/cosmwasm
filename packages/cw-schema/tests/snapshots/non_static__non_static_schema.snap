---
source: packages/cw-schema/tests/non_static.rs
expression: schema
---
{
  "type": "v1",
  "root": 0,
  "definitions": [
    {
      "name": "non_static_NonStatic",
      "type": "struct",
      "properties": {
        "test1": {
          "value": 1
        },
        "test2": {
          "value": 1
        },
        "test3": {
          "value": 1
        },
        "test4": {
          "value": 1
        }
      }
    },
    {
      "name": "str",
      "type": "string"
    }
  ]
}
